src/lib/react-query/maps.ts

```diff
@@ -36,3 +122,119 @@ export function mapStyleJsonUrlQueryOptions({
 		},
 	})
 }
+
+export function getMapSharesQueryOptions({
+	clientApi,
+}: {
+	clientApi: MapeoClientApi
+}) {
+	return queryOptions({
+		...baseQueryOptions(),
+		queryKey: getMapSharesQueryKey(),
+		queryFn: async (): Promise<Array<MapShare>> => {
+			return [MOCK_MAP_SHARE]
+		},
+	})
+}
+
+export function getMapShareByIdQueryOptions({
+	clientApi,
+	shareId,
+}: {
+	clientApi: MapeoClientApi
+	shareId: string
+}) {
+	return queryOptions({
+		...baseQueryOptions(),
+		queryKey: getMapSharesByIdQueryKey({ shareId }),
+		queryFn: async (): Promise<MapShare> => {
+			return MOCK_MAP_SHARE
+		},
+	})
+}
+
+export function acceptMapShareMutationOptions({
+	clientApi,
+	queryClient,
+}: {
+	clientApi: MapeoClientApi
+	queryClient: QueryClient
+}) {
+	return {
+		...baseMutationOptions(),
+		mutationFn: async ({ shareId }) => {
+			await new Promise((res) => setTimeout(res, 1000))
+			console.log('Accepted map share', shareId)
+		},
```

Comment:
i think this should return the shareId when resolved. In order to show the progress of the map being downloaded, we need to `useSingleMapShare` which requires the shareId, so it would be good to pass it directly from this function